core/App/components/inputs/PinInput.tsx

```diff
@@ -0,0 +1,99 @@
+import React, { useState } from 'react'
+import { Keyboard, StyleSheet, Text, TouchableOpacity, View } from 'react-native'
+import { CodeField, useClearByFocusCell } from 'react-native-confirmation-code-field'
+import Icon from 'react-native-vector-icons/MaterialIcons'
+
+import { useTheme } from '../../contexts/theme'
+
+const CELL_COUNT = 6
+
+interface PinInputProps {
+  label?: string
+  onPinChanged?: (pin: string) => void
+}
+
+const PinInput: React.FC<PinInputProps> = ({ label, onPinChanged }) => {
+  const [pin, setPin] = useState('')
+  const [showPin, setShowPin] = useState(false)
+  const [props, getCellOnLayoutHandler] = useClearByFocusCell({
+    value: pin,
+    setValue: setPin,
+  })
+
+  const { TextTheme, PinInputTheme } = useTheme()
+  const style = StyleSheet.create({
+    codeField: {
+      flexDirection: 'row',
+      justifyContent: 'space-between',
+      alignItems: 'center',
+    },
+    codeFieldRoot: {
+      marginBottom: 24,
+    },
+    cell: {
+      width: 40,
+      height: 48,
+      backgroundColor: PinInputTheme.cell.backgroundColor,
+      borderWidth: 2,
+      borderRadius: 5,
+      borderColor: PinInputTheme.cell.borderColor,
+      marginRight: 8,
+    },
+    focusedCell: {
+      borderColor: PinInputTheme.focussedCell.borderColor,
+    },
+    cellText: {
+      ...TextTheme.headingThree,
+      color: PinInputTheme.cellText.color,
+      textAlign: 'center',
+      textAlignVertical: 'center',
+      lineHeight: 42,
+    },
+  })
+
+  return (
+    <View>
+      {label && <Text style={[TextTheme.label, { marginBottom: 8 }]}>{label}</Text>}
+      <View style={[style.codeField]}>
+        <CodeField
+          {...props}
+          value={pin}
+          onChangeText={(value: string) => {
+            onPinChanged && onPinChanged(value)
+            setPin(value)
+            if (value.length === 6) {
```

Comment:
Magic number? Probably meant to be `CELL_COUNT`

Reply:
Using exported constant